internal/fs/inode/file.go

```diff
@@ -623,10 +673,15 @@ func (f *FileInode) SetMtime(
 // LOCKS_REQUIRED(f.mu)
 func (f *FileInode) Sync(ctx context.Context) (err error) {
 	// If we have not been dirtied, there is nothing to do.
-	if f.content == nil {
+	if f.content == nil && f.bwh == nil {
 		return
 	}
 
+	if f.bwh != nil {
+		// Finalize the object.
+		return f.flushBufferedWriteHandlerAndUpdateInode()
```

Comment:
we shouldn't flush on sync call.
from LLD:

> Sync() can be called multiple times during the lifetime of the file write. Once the resumable upload is finalized we will not be able to append anymore data to it. But the intention of sync() call from the application side, is to persist all the data to the backend device. So we will block the syncFile() call and upload all pending buffers to GCS and free up the free buffers. We will not finalize the upload.

Reply:
Hey, I wanted to follow this. Is it possible to get the access to the LLD doc?

Reply:
Hey, I wanted to follow this. Is it possible to get access to the LLD?